### PR TITLE
Add unit tests to self host F4E

### PR DESCRIPTION
**Pull request by justindbaur**
## Type of change
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Add more test coverage to self hosted families for enterprise additions.

## Code changes

* **src/Core/OrganizationFeatures/OrganizationSponsorships/FamiliesForEnterprise/Cloud/CloudSyncSponsorshipsCommand.cs:** Move two checks pertaining to the `sponsoringOrg` to outside of the loop so they don't have to be done for each sponsorship.

## Testing requirements

N/A

## Before you submit
- [x] I have checked for formatting errors (`dotnet tool run dotnet-format --check`) (required)
- [ ] If making database changes - I have also updated Entity Framework queries and/or migrations
- [x] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

File: src/Core/OrganizationFeatures/OrganizationSponsorships/FamiliesForEnterprise/Cloud/CloudSyncSponsorshipsCommand.cs [hunk 1]
```diff
@@ -54,11 +54,13 @@ private async Task<(IEnumerable<OrganizationSponsorshipData> data, IEnumerable<O
             var sponsorshipIdsToDelete = new List<Guid>();
             var sponsorshipsToReturn = new List<OrganizationSponsorshipData>();
 
+            var sponsoringOrganizationProduct = StaticStore.GetPlan(sponsoringOrg.PlanType).Product;
+            var sponsoringOrgDisabledForMoreThanGracePeriod = OrgDisabledForMoreThanGracePeriod(sponsoringOrg);
+
```
Comment (MGibson1) [inline src/Core/OrganizationFeatures/OrganizationSponsorships/FamiliesForEnterprise/Cloud/CloudSyncSponsorshipsCommand.cs:59]: These both have very little value, they're just lookups on objects in memory and a little calculation. I'm not sure it's worth the extra lines needed to store this

**Review by MGibson1 — CHANGES_REQUESTED** [review]

File: src/Core/OrganizationFeatures/OrganizationSponsorships/FamiliesForEnterprise/Cloud/CloudSyncSponsorshipsCommand.cs (hunk 1 above)
Comment (justindbaur) [reply src/Core/OrganizationFeatures/OrganizationSponsorships/FamiliesForEnterprise/Cloud/CloudSyncSponsorshipsCommand.cs]: Alright, reverted.

File: test/Core.Test/OrganizationFeatures/OrganizationSponsorships/FamiliesForEnterprise/Cloud/CloudSyncSponsorshipsCommandTests.cs [hunk 2]
```diff
@@ -88,5 +88,94 @@ await sutProvider.GetDependency<IOrganizationSponsorshipRepository>()
...
+                .DeleteManyAsync(Arg.Is<IEnumerable<Guid>>(toDeleteIds =>
+                    toDeleteIds.Count() == 1 && toDeleteIds.ElementAt(0) == canDeleteSponsorship.Id));
+        }
+
+        [Theory]
+        [BitAutoData]
+        public async Task SyncOrganization_BadData_DoesNotSave(SutProvider<CloudSyncSponsorshipsCommand> sutProvider,
+            Organization sponsoringOrganization, OrganizationSponsorship badOrganizationSponsorship)
+        {
+            sponsoringOrganization.PlanType = PlanType.EnterpriseAnnually;
```
Comment (jlf0dev) [inline test/Core.Test/OrganizationFeatures/OrganizationSponsorships/FamiliesForEnterprise/Cloud/CloudSyncSponsorshipsCommandTests.cs:178]: Does this test do anything?

Comment (justindbaur) [reply test/Core.Test/OrganizationFeatures/OrganizationSponsorships/FamiliesForEnterprise/Cloud/CloudSyncSponsorshipsCommandTests.cs:178]: It did not, I have added the test I meant to be there.

**Review by jlf0dev — APPROVED** [review]

**Pull request merged**